incompatibility/incompatibility-report/chain_test.go

```diff
@@ -0,0 +1,124 @@
+// Copyright 2021 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package incompatibility
+
+import (
+	"testing"
+
+	openapiv3 "github.com/googleapis/gnostic/openapiv3"
+)
+
+// incompatibilityEquality checks equality for two incompatibilities
+func incompatibilityEquality(i1 *Incompatibility, i2 *Incompatibility) (equality bool) {
```

Comment:
There are a number of packages used for equality testing.    I'm not sure which one is most often used, and perhaps there is one already used for this project. 

Equality facilities also often support returning a representation of the diff.

Reply:
now using diffs